src/restconfig/src/main/java/org/geoserver/catalog/rest/DataStoreResource.java

```diff
@@ -179,6 +180,16 @@ protected void configurePersister(XStreamPersister persister, DataFormat format)
         persister.setCallback( 
             new XStreamPersister.Callback() {
                 @Override
+                protected CatalogInfo getCatalogObject() {
```

Comment:
You are having to override this method in a lot of classes (DataStoreResource, FeatureTypeResource, CoverageStoreResource, CoverageResource) - what was wrong with the previous implementation?